### ISSUE TITLE
Question on reward debiasing

### ISSUE THREAD

**Issue by JefferyChen453** [OpenBMB/RLPR#10]
Thanks for your great work.

I observe that scoreA is computed before the training loop: 

https://github.com/OpenBMB/RLPR/blob/main/verl/trainer/ppo/ray_trainer.py#L1509-L1511

which means the debiased score is computed by updated scoreB and fixed scoreA. Is there any insight for not updating scoreA?

btw can you share the scoreB & score A curve during training? I'm trying to implement RLPR in my specific task and found scoreB and scoreA usually has merely a minor gap.

**Comment by Ma-Weijian** [comment]
I trained the model on 4 A100-80Gs. There does have a gap, but score A is lower than score B.

**Comment by yiranyyu** [comment]
> Thanks for your great work.
> 
> I observe that scoreA is computed before the training loop:
> 
> https://github.com/OpenBMB/RLPR/blob/main/verl/trainer/ppo/ray_trainer.py#L1509-L1511
> 
> which means the debiased score is computed by updated scoreB and fixed scoreA. Is there any insight for not updating scoreA?
> 
> btw can you share the scoreB & score A curve during training? I'm trying to implement RLPR in my specific task and found scoreB and scoreA usually has merely a minor gap.

We fix scoreA to save the computation cost and empirically found this has limited influence to performance.

As for training log, we did not log the origin scoreA. Here we can share the before-filter scoreB log and before-filter reward log (filtering can directly change the score distribution) for a recent analytic experiments using 20% our training dataset and Qwen2.5 3B. Hope it could be helpful.


<img width="313" height="221" alt="Image" src="https://github.com/user-attachments/assets/660a68c0-17d7-49a4-83e7-2d8bd7b87e62" />

<img width="983" height="242" alt="Image" src="https://github.com/user-attachments/assets/0ec7a6be-2a8f-4bff-baaa-86b0d0e19f5a" />

**Comment by JefferyChen453** [comment]
Thanks for sharing!
